### PR TITLE
Extend documentation

### PR DESCRIPTION
**Pull request by nagyrobi**
Add practical information about how to configure with multiple DMX fixtures on the same bus, including an extensive example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # esphome-dmx512
 
-This is a DMX512 custom component for ESPHome that allows Arduino-based
-devices to control DMX devices via UART (this requires usually an RS485 module). ESP32 with the IDF-framework is currently not supported.
+This is a DMX512 custom component for ESPHome that allows Arduino-based devices to control DMX devices via UART (this requires an RS485 module, see below). ESP32 with the IDF-framework is currently not supported.
```
Comment (andyboeh) [inline README.md:4]: Please do not remove the line breaks!

File: README.md [hunk 2]
```diff
@@ -10,18 +9,17 @@ The implementation is based on other projects:
   * https://github.com/cupertinomiranda/esphome
   * https://github.com/jakosek/esphome/tree/dmx_no_uart
 
-It differs in that it uses the internal UART component but generates the
-break signal by detaching and re-attaching the GPIO pin (on ESP32).
+It differs in that it uses the internal UART component but generates the break signal by detaching and re-attaching the GPIO pin.
```
Comment (andyboeh) [inline README.md:13]: the ESP32 statement in brackets is on purpose. The signal is generated differently on the ESP8266. I suggest the removal of this sentence as it is not really necessary.

File: README.md [hunk 3]
```diff
@@ -49,4 +47,31 @@ dmx512:
   * `force_full_frames`: If set to true, the full 513-byte frame is always sent. Otherwise, only the configured channels are transmitted.
   * `custom_mab_len`: Set a custom mark-after-break length (in uS, default 12)
   * `custom_break_len`: Set a custom break length (in uS, default 92)
-  * `update_interval`: Specify a custom update interval, i.e. the minimum time between resending the current values (in ms, default 500)
\ No newline at end of file
+  * `update_interval`: Specify a custom update interval, i.e. the minimum time between resending the current values (in ms, default 500)
+
+Outputs point to channels in the DMX universe, from 1 to 512. Some DMX fixtures can be daisy chained with XLR cables, and each one can be set to operate at a different address. When you specify the output channel in the configuration of this component, take into account that _channel (in esphome component) = address + channel (on the fixture)_. When you choose addresses for your DMX fixtures keep in mind that the next free address number on the bus that can be assigned is the _address of the previous light + the last channel number of that light_. 
+
```
Comment (andyboeh) [inline README.md:53]: Please add line breaks in this and the following paragraphs.

File: README.md [hunk 4]
```diff
@@ -49,4 +47,31 @@ dmx512:
...
+For easeier understanding, eg. the [ADJ VBar Pak](https://d295jznhem2tn9.cloudfront.net/ItemRelatedFiles/8659/vbar_pak.pdf) can be used in various modes. When you have 4 of this daisy chained, if you set them up to 8-channel mode (to access all the effects) you can set the address of the first one 1, the next one to 9, the third one to 17, fourth one to 25, using the buttons on each fixture. Then in the config you'd have channels for the first one from 1 to 8, the second one from 9 to 16, third one from 17 to 24, forth one from 25 to 32. See [another example](example_4x_adj_vbar_pak.yaml) for this kind of configuration.
+
```
Comment (andyboeh) [inline README.md:55]: easier
... 4 of these...
...of the first one to 1...

File: README.md [hunk 5]
```diff
@@ -49,4 +47,31 @@ dmx512:
...
+The `output` implements float values between 0 and 100. You can use it not only for `light`, but for any component which can send data to it, moreover, it can be set from lambdas. The above example shows use cases for this, where certain outputs are used for hardware effects selection on the DMX fixture, using a `select` component, and parameters can be adjusted from `number` entities.
+
+## Wiring
+
+You can use an RS485-TTL adapter module to connect your ESP devce with the DMX bus.
+
```
Comment (andyboeh) [inline README.md:61]: device

File: README.md [hunk 6]
```diff
@@ -49,4 +47,31 @@ dmx512:
...
+MAX485-M GND     -> ESP GND
+MAX485-M RE & DE -> ESP +3.3V
+MAX485-M DI      -> ESP32 GPIO5 or ESP8266 GPIO2 (as per examples above)
+MAX485-M A       -> XLR 3 (DMX +)
+MAX485-M B       -> XLR 2 (DMX -)
+MAX485-M GND     -> XLR 1 (DMX GND)
+```
+
+Don't forget about 120Ohm termination resistors. If your fixture has DMX IN and OUT ports, on the OUT port or the last fixture in the chain you should use a termination resistor between XLR pins 2 and 3. Similarly on MAX485-M, it has to be placed in parallel with A and B outputs, given that it's going to be placed at the start of the chain.     
+Using good quality 120Ohm impedance cables, DMX lines can be run a maximum distance of approximately 1000 metres. With CAT5 cable DMX lines are safe until approximately 300 meters.
```
Comment (andyboeh) [inline README.md:77]: Are you sure about 1000 metres?

Comment (andyboeh) [reply README.md:77]: Does the MAX really support 3.3V logic when powered by 5V? I would suggest the MAX3485 instead which is directly powered by 3.3V.

File: example_4x_adj_vbar_pak.yaml [hunk 7]
```diff
@@ -0,0 +1,504 @@
+# Example configuration for four ADJ VBar Pak fixtures daisy chained with XLR cables
+# DMX description: https://d295jznhem2tn9.cloudfront.net/ItemRelatedFiles/8659/vbar_pak.pdf
+# Fixture configuration set for DMX 8 channel mode
+
+uart:
+  id: uart_bus
+  baud_rate: 250000
+  tx_pin: 2 #ESP8266
```
Comment (andyboeh) [inline example_4x_adj_vbar_pak.yaml:8]: GPIO2

File: example_4x_adj_vbar_pak.yaml [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+  stop_bits: 2
+
+dmx512:
+  id: dmx
+  uart_id: uart_bus
+  tx_pin: 2 #ESP8266
```
Comment (andyboeh) [inline example_4x_adj_vbar_pak.yaml:14]: GPIO2

File: example_4x_adj_vbar_pak.yaml [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+  uart_num: 1
+  force_full_frames: true
```
Comment (andyboeh) [inline example_4x_adj_vbar_pak.yaml:16]: That's the default, remove this line.

File: example_4x_adj_vbar_pak.yaml [hunk 10]
```diff
@@ -0,0 +1,504 @@
...
+  min_value: 0
+  max_value: 255
+  step: 1
+  mode: box
+  restore_value: true
+  set_action:
+    - lambda: |-
+        id(vbar4_effect_mode).set_level(x / 255);
+
+
```
Comment (andyboeh) [inline example_4x_adj_vbar_pak.yaml:261]: Remove empty lines

File: example_4x_adj_vbar_pak.yaml [hunk 11]
```diff
@@ -0,0 +1,504 @@
...
+  options:
+    - "None"
+    - "Color Macro"
+    - "Color Change"
+    - "Color Fade"
+    - "Sound Active"
+    - "Strobing"
+  set_action:
+    - lambda: |-
+          if (0 == x.compare(std::string{"None"})) {
```
Comment (andyboeh) [inline example_4x_adj_vbar_pak.yaml:279]: if(x.compare() == 0)

Otherwise, it doesn't match the coding style of the component.

**Review by andyboeh — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (nagyrobi) [reply README.md:99]: Empirically tested - works. Powered at either 5 or 3.3V.

Comment (nagyrobi) [reply README.md:99]: > Are you sure about 1000 metres?

Yes, with proper cables. I work in the industry.

File: example_4x_adj_vbar_pak.yaml (hunk 9 above)
Comment (nagyrobi) [reply example_4x_adj_vbar_pak.yaml]: added that info to the readme

Comment (andyboeh) [reply example_4x_adj_vbar_pak.yaml]: Sorry, it's not the default (I fixed that), but I do not recommend it.

**Pull request merged**